### PR TITLE
📖Subscriptions: Adding analytics documentation

### PR DESCRIPTION
**Pull request by prateekbh**

### CODE REVIEW

File: extensions/amp-subscriptions/amp-subscriptions.md [hunk 1]
```diff
@@ -348,6 +348,36 @@ For instance, to show a "subscribe" action to non-subscribers:
 <button subscriptions-display="NOT subscribed" subscriptions-action="subscribe">Become a subscriber</button>
 ```
 
+## Analytics
+
+The `amp-subscriptions` triggers seven types of analytics signals during the different lifecycle of various subscriptions.
+Following are the events and the conditions when the events are triggered.
+
+1. PLATFORM_ACTIVATED:
```
Comment (dvoytenko) [inline extensions/amp-subscriptions/amp-subscriptions.md]: Use the actual values that can be used in the analytics. E.g. "subscriptions-platform-activated".

File: extensions/amp-subscriptions/amp-subscriptions.md [hunk 2]
```diff
@@ -348,6 +348,36 @@ For instance, to show a "subscribe" action to non-subscribers:
...
+5. ACTION_DELEGATED
+ - A platform can request its action to be delegated to another service, in such scenerio `ACTION_DELEGATED` is triggered just before handing over the event to the other platform.
+ - This event is fired with `serviceId` and `action` of the selected platform.
+6. ENTITLEMENT_RESOLVED
+ - Once the platform is registered, the entitlments from it are requested. `ENTITLEMENT_RESOLVED` event is triggered when the entitlement fetch from the platform is completed.
+ - This event is fired with `serviceId` and `action` of the selected platform.
+7. STARTED
+ - This event is triggered when `amp-subscriptions` is initialized.
+ - This event does not contain any data.
+8. `<ActionName>-`STARTED
```
Comment (dvoytenko) [inline extensions/amp-subscriptions/amp-subscriptions.md]: it's actually "subscriptions-action-ActionName-started"

File: extensions/amp-subscriptions/amp-subscriptions.md [hunk 3]
```diff
@@ -348,6 +348,36 @@ For instance, to show a "subscribe" action to non-subscribers:
...
+ - Whenever any action execution starts, And event with `<ActionName>-`STARTED is fired.
```
Comment (dvoytenko) [inline extensions/amp-subscriptions/amp-subscriptions.md]: Similarly, there are also "subscriptions-action-ActionName-success" and "-failed".

File: extensions/amp-subscriptions/amp-subscriptions.md (hunk 1 above)
Comment (prateekbh) [reply extensions/amp-subscriptions/amp-subscriptions.md]: Done

File: extensions/amp-subscriptions/amp-subscriptions.md (hunk 3 above)
Comment (prateekbh) [reply extensions/amp-subscriptions/amp-subscriptions.md]: Done

File: extensions/amp-subscriptions/amp-subscriptions.md (hunk 2 above)
Comment (prateekbh) [reply extensions/amp-subscriptions/amp-subscriptions.md]: Done

**Comment by prateekbh** [comment]
@rsimha should code cov, report for changes in `.md` or in `examples` at all?

**Comment by prateekbh** [comment]
Oh ok, I didn't know about unit tests using files in `examples/`. We should be good then. Thanks 👍

**Comment by rsimha** [comment]
Yes, they are used. 

https://github.com/ampproject/amphtml/blob/5446179ef0e79808807edc59b4dda6f2dd8d622d/build-system/config.js#L59

**Comment by prateekbh** [comment]
@dvoytenko merging this, so that publishers can see the details. Still, feel free to comment on anything and i'll send a follow up PR.

**Pull request merged**